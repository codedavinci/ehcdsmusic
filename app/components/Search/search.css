.search-wrapper {
  display: flex;
  width: 100%;
  padding: 1.2em 0;
}

.search-wrapper form {
  margin: auto;
}

.search-wrapper input {
  border-radius: 0.5em;
  height: 1.9em;
  width: 16em;
  border: 1px solid lightgray;
  font-family: 'Lato', 'sans-serif';
  font-size: 26px;
  padding-left: 0.8em;
  color: gray;
  margin: 25px 10px 10px 0;
  transition: all 0.2s ease-in-out;
  -webkit-box-shadow: -1px 9px 74px -12px rgba(0, 0, 0, 0.75);
  -moz-box-shadow: -1px 9px 74px -12px rgba(0, 0, 0, 0.75);
  box-shadow: -1px 9px 74px -12px rgba(0, 0, 0, 0.75);
}

.search-wrapper input:focus {
  border: 1px solid #a9a9a9;
}

.search-wrapper button {
  height: 3.2em;
  width: 7em;
  border-radius: 1em;
  background-color: #000;
  color: #fff;
  font-family: 'Lato', 'sans-serif';
  font-weight: 400;
  font-size: 110%;
  letter-spacing: 1px;
  padding-top: 9px;
  transition: 0.2s ease-in-out;
}

.search-wrapper button:focus,
input:focus {
  outline: none;
}

.search-wrapper button:hover {
  box-shadow: 0px 17px 10px -10px rgba(0, 0, 0, 0.4);
}
