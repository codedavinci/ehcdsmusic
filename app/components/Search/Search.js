import React, { Component } from 'react'
import './search.css'

import PropTypes from 'prop-types'

const Search = ({ onSubmitSearch }) => {
  let Input

  const onSubmit = e => {
    e.preventDefault()
    if (Input.value) {
      onSubmitSearch(Input.value)
    }
  }

  return (
    <div className="search-wrapper">
      <form onSubmit={onSubmit}>
        <input type="text" ref={e => (Input = e)} />
        <button type="submit">SEARCH</button>
      </form>
    </div>
  )
}

Search.propType = {
  onSubmitSearch: PropTypes.func.isRequired,
}

export default Search
