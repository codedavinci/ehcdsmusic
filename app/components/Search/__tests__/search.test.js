import React from 'react'

import Search from '../Search'

describe('<Search />', () => {
  const defaultProps = {
    handleSearch: jest.fn(),
  }

  const doShallow = (overrides = {}) => {
    const search = shallow(<Search {...overrides} />)

    return search
  }

  it('renders', () => {
    const search = doShallow(defaultProps)
    expect(search).toMatchSnapshot()
  })
})
