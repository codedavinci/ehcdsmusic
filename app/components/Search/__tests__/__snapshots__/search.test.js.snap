// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    onSubmitSearch={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        onSubmit={[Function]}
      >
        <input
          type="text"
        />
        <button
          type="submit"
        >
          SEARCH
        </button>
      </form>,
      "className": "search-wrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            type="text"
          />,
          <button
            type="submit"
          >
            SEARCH
          </button>,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "type": "text",
          },
          "ref": [Function],
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SEARCH",
            "type": "submit",
          },
          "ref": null,
          "rendered": "SEARCH",
          "type": "button",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          onSubmit={[Function]}
        >
          <input
            type="text"
          />
          <button
            type="submit"
          >
            SEARCH
          </button>
        </form>,
        "className": "search-wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              type="text"
            />,
            <button
              type="submit"
            >
              SEARCH
            </button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "type": "text",
            },
            "ref": [Function],
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "SEARCH",
              "type": "submit",
            },
            "ref": null,
            "rendered": "SEARCH",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
